sdk/nodejs/runtime/resource.ts

```diff
@@ -84,9 +88,11 @@ interface ResourceResolverOperation {
     // all be URNs of custom resources, not component resources.
     propertyToDirectDependencyURNs: Map<string, Set<URN>>;
     // A list of aliases applied to this resource.
-    aliases: URN[];
+    aliases: (Alias | URN)[];
     // An ID to import, if any.
     import: ID | undefined;
+    // Any important feature support from the monitor.
+    monitorSupports: Map<string, boolean>;
```

Comment:
I'd make this an explicit bool flag about the state of "aliases" (i.e. are they pre-expanded or not) rather than passing string maps around.